[15기 김원희] ADD: 로그인 , 인증 데코레이터 기능 및 테스트 구현

supersaloon opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
email 계정으로 가입한 계정에 대해 로그인 기능 및 테스트 구현
		 
## 수정 사항들 자세한 내용
- 로그인 기능 및 테스트 구현
  - 회원가입된 email 및 password와 일치하는 값을 전달받으면 로그인 처리
    - status=200 및 ACCESS_TOKEN 리턴
  - 존재하지 않는 email로 로그인 시도하면 400 리턴
  - email은 존재하나 password가 일치하지 않으면 400 리턴
- 로그인 데코레이터 구현

## 기타 질문 및 특이 사항


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.
- [x] Git rebase와 squash를 했고 커밋 수가 하나 인것을 확인 했습니다.

lordmyshepherd commented on tests/test_user_signin.py:
```diff
@@ -0,0 +1,58 @@
+
+    def test_signin_success(self):
+        client = Client()
+        response = client.post('/user/signin', json.dumps(SIGNIN_USER), content_type='application/json')
```
`SIGNIN_USER` 데이터는 어떤 데이터 인가요?
```suggestion
        response = client.post('/user/signin', json.dumps(SIGNIN_USER), content_type='application/json')
```

supersaloon replied:
반복되는 코드라서 test 밖의 파일에 빼 두었던 변수입니다.
setUp에 담아서 불러오도록 수정하였습니다.

lordmyshepherd commented on tests/test_user_signup.py:
```diff
@@ -0,0 +1,118 @@
+from user.models import Authentication, User, AccountType
+from user.utils  import SIGNUP_USER
+
+
```
이전 pr에서 올라온 코드 같은데 또 올라왔네요??

supersaloon replied:
이전 PR에 있던 파일은 현재 PR에서 없애도록 하겠습니다.

supersaloon committed:
[Add] feature/user-signin: 로그인 기능 및 테스트 구현
- 로그인 기능 및 테스트 구현
  - 회원가입된 email 및 password와 일치하는 값을 전달받으면 로그인 처리
    - status=200 및 ACCESS_TOKEN 리턴
  - 존재하지 않는 email로 로그인 시도하면 400 리턴
  - email은 존재하나 password가 일치하지 않으면 400 리턴

  - PR#5 리뷰 반영하여 수정
    - 이전 PR에 해당되는 파일 삭제
    - 반복되는 코드는 setUp에 담아서 사용
    - signup 관련 코드 삭제
    - utils.py 공백 삭제

  - 로그인 데코레이터 추가

[Modify] 데코레이터 구현 후 conflict 해결

lordmyshepherd commented:
LGTM 👍
